### PR TITLE
Allow params to be passed to get requests

### PR DESCRIPTION
**Pull request by prpetten**
Need to pass since_id to `client.mentions.all` request

### CODE REVIEW

File: lib/name_drop/client.rb [hunk 1]
```diff
@@ -66,23 +66,27 @@ def post(endpoint, attributes)
...
+      make_request(endpoint, 'Delete')
     end
 
     # @!endgroup
 
     private
 
+    def make_request(endpoint, method, attributes = {})
+      request = "Net::HTTP::#{method}".constantize.new request_uri(endpoint)
+      request.body = JSON.dump(attributes) unless attributes.empty?
```
Comment (ridiculous) [inline lib/name_drop/client.rb]: How about `if attributes.any?`

File: lib/name_drop/client.rb [hunk 2]
```diff
@@ -66,23 +66,27 @@ def post(endpoint, attributes)
...
+      execute_request request
```
Comment (ridiculous) [inline lib/name_drop/client.rb]: I kinda like the functional composure style. What do you think about this just returning the request and the callers (`delete`, `put`, etc) pass it to execute

``` ruby
execute_request build_request(endpoint, 'Put', attributes)`)
```

**Review by ridiculous — COMMENTED** [review]
Looks good! Just a few comments

File: lib/name_drop/client.rb (hunk 2 above)
Comment (prpetten) [reply lib/name_drop/client.rb]: I like it

File: lib/name_drop/client.rb (hunk 1 above)
Comment (prpetten) [reply lib/name_drop/client.rb]: Yeah I like `if` better than `unless`

File: lib/name_drop/client.rb [hunk 3]
```diff
@@ -66,23 +66,27 @@ def post(endpoint, attributes)
...
+      execute_request build_request(endpoint, 'Delete')
     end
 
     # @!endgroup
 
     private
 
+    def build_request(endpoint, method, attributes = {})
+      request = "Net::HTTP::#{method}".constantize.new request_uri(endpoint)
+      request.body = JSON.dump(attributes) if attributes.any?
```
Comment (ridiculous) [inline lib/name_drop/client.rb:86]: I thought you needed to set the query params for a GET request on the URI object, like in the [docs](http://ruby-doc.org/stdlib-2.3.1/libdoc/net/http/rdoc/Net/HTTP.html), but it looks like this also works, just that params aren't present in the URL.

**Comment by ridiculous** [comment]
Added a client spec in the last commit

**Pull request merged**